internal/provider/cluster_resource.go

```diff
@@ -72,194 +76,110 @@ func (r clusterResourceType) GetSchema(_ context.Context) (tfsdk.Schema, diag.Di
-										Optional: true,
-									},
-									"disk_iops": {
-										Type:     types.Int64Type,
-										Optional: true,
-									},
-									"machine_spec": {
-										Optional: true,
-										Attributes: tfsdk.SingleNestedAttributes(map[string]tfsdk.Attribute{
-											"machine_type": {
-												Type:     types.StringType,
-												Optional: true,
-											},
-											"num_virtual_cpus": {
-												Type:     types.Int64Type,
-												Optional: true,
-											},
-										}),
-									},
-								}),
-							},
-						}),
+						Computed: true,
+						PlanModifiers: tfsdk.AttributePlanModifiers{
+							tfsdk.UseStateForUnknown(),
+						},
 					},
-				}),
-			},
-			"config": {
-				Computed: true,
-				PlanModifiers: tfsdk.AttributePlanModifiers{
-					tfsdk.UseStateForUnknown(),
-				},
-				Attributes: tfsdk.SingleNestedAttributes(map[string]tfsdk.Attribute{
-					"serverless": {
+					"memory_gib": {
+						Computed: true,
+						Type:     types.Float64Type,
+						PlanModifiers: tfsdk.AttributePlanModifiers{
+							tfsdk.UseStateForUnknown(),
+						},
+					},
+					"machine_type": {
+						Type:     types.StringType,
+						Optional: true,
 						Computed: true,
 						PlanModifiers: tfsdk.AttributePlanModifiers{
 							tfsdk.UseStateForUnknown(),
 						},
-						Attributes: tfsdk.SingleNestedAttributes(map[string]tfsdk.Attribute{
-							"spend_limit": {
-								Computed: true,
-								Type:     types.Int64Type,
-								PlanModifiers: tfsdk.AttributePlanModifiers{
-									tfsdk.UseStateForUnknown(),
-								},
-							},
-							"routing_id": {
-								Computed: true,
-								Type:     types.StringType,
-								PlanModifiers: tfsdk.AttributePlanModifiers{
-									tfsdk.UseStateForUnknown(),
-								},
-							},
-						}),
 					},
-					"dedicated": {
+					"num_virtual_cpus": {
+						Optional: true,
 						Computed: true,
+						Type:     types.Int64Type,
 						PlanModifiers: tfsdk.AttributePlanModifiers{
 							tfsdk.UseStateForUnknown(),
 						},
-						Attributes: tfsdk.SingleNestedAttributes(map[string]tfsdk.Attribute{
-							"machine_type": {
-								Computed: true,
-								Type:     types.StringType,
-							},
-							"num_virtual_cpus": {
-								Computed: true,
-								Type:     types.Int64Type,
-							},
-							"storage_gib": {
-								Computed: true,
-								Type:     types.Int64Type,
-							},
-							"memory_gib": {
-								Computed: true,
-								Type:     types.Float64Type,
-							},
-							"disk_iops": {
-								Computed: true,
-								Type:     types.Int64Type,
-							},
-						}),
 					},
 				}),
 			},
 			"regions": {
-				Computed: true,
+				Required: true,
 				PlanModifiers: tfsdk.AttributePlanModifiers{
 					tfsdk.UseStateForUnknown(),
 				},
 				Attributes: tfsdk.ListNestedAttributes(map[string]tfsdk.Attribute{
 					"name": {
-						Computed: true,
+						Required: true,
 						Type:     types.StringType,
 					},
```

Comment:
Earlier we had `create_spec` and `update_spec` differently. `update_spec` has the only fields which are allowed to be updated in cluster. However now we only have `serverless` and `dedicated` field, hence we have to add the validation which are not allowed to be updated. For example: the `region` is not allowed to be updated. Our TF provider should properly throw error that you are not allowed to be updated.  

TF provider is crashing when we try to update the region:
```
cockroach_cluster.cockroach: Modifying... [id=d161cece-663a-448e-964e-c4217db6fe9c]
╷
│ Error: Provider produced inconsistent result after apply
│ 
│ When applying changes to cockroach_cluster.cockroach, provider "provider[\"registry.terraform.io/cockroachdb/cockroach\"]" produced an unexpected new
│ value: .regions[0].name: was cty.StringVal("us-east2"), but now cty.StringVal("us-east1").
│ 
│ This is a bug in the provider, which should be reported in the provider's own issue tracker.
╵
```